swank/abcl.lisp

```diff
@@ -541,12 +541,19 @@
 (defmethod source-location ((symbol symbol))
   (when (pathnamep (ext:source-pathname symbol))
     (let ((pos (ext:source-file-position symbol)))
-      `(:location
-        (:file ,(namestring (ext:source-pathname symbol)))
-        ,(if pos
-             (list :position (1+ pos))
-             (list :function-name (string symbol)))
-        (:align t)))))
+      (if (ext:pathname-jar-p (ext:source-pathname symbol))
+          (destructuring-bind (jar entry) (abcl-asdf::split-string (subseq (namestring (ext:source-pathname symbol)) 9) (char "!" 0))
```

Comment:
split-string -> brain fart.
Subseq: Will add comment.
```
(describe #P"jar:file:/Users/alanr/repos/lsw/lib/abcl-contrib.jar!/jss/invoke.lisp")
#P"jar:file:/Users/alanr/repos/lsw/lib/abcl-contrib.jar!/jss/invoke.lisp" is an object of type JAR-PATHNAME:
  HOST         NIL
  DEVICE       (#P"/Users/alanr/repos/lsw/lib/abcl-contrib.jar")
  DIRECTORY    (:ABSOLUTE "jss")
  NAME         "invoke"
  TYPE         "lisp"
  VERSION      NIL
```
Using the pathname manipulations wouldn't be clearer since this is a nonstandard extension. I don't feel bad having this in the abcl-specific code.